vllm/model_executor/layers/mamba/mamba_mixer.py

```diff
@@ -421,16 +431,25 @@ def split_batch_to_prefill_and_decode(
     num_decode_tokens: int,
     num_prefills: int,
     num_decodes: int,
+    num_padded_decodes: int,
 ) -> PrefillDecodeSplit:
+    num_actual_tokens = num_prefill_tokens + num_decode_tokens
+
     if envs.VLLM_USE_V1:
         # In v1, decode tokens come first, then prefill tokens.
         hidden_states_BC_d, hidden_states_BC_p = torch.split(
-            hidden_states_BC, [num_decode_tokens, num_prefill_tokens], dim=-1)
-        gate_d, gate_p = torch.split(gate,
+            hidden_states_BC[..., :num_actual_tokens],
+            [num_decode_tokens, num_prefill_tokens],
+            dim=-1)
+        gate_d, gate_p = torch.split(gate[..., :num_actual_tokens],
                                      [num_decode_tokens, num_prefill_tokens],
                                      dim=-1)
+
+        # num_padded_decodes accounts for CUDA graph padding when applicable
         state_indices_tensor_d, state_indices_tensor_p = torch.split(
-            state_indices_tensor, [num_decodes, num_prefills], dim=0)
+            state_indices_tensor[:num_padded_decodes + num_prefills],
+            [num_decodes, num_prefills],
```

Comment:
Not sure I fully get this. The padding should only happen when FCG is enabled, which right now only happens for decode-only batches. Therefore, in that scenario `num_prefill=0`. In that case, can't `num_decodes` be strictly less than `num_padded_decodes` which would cause this to fail? Shouldn't we split it to `[num_padded_decodes, num_prefills]`?